### DISCUSSION TITLE
Compilation problems on Windows

### DISCUSSION THREAD

**Discussion by tannenfreund87** [librespot-org/librespot#1400] [General]
I don't want to open a bug report, as I think the problem is on layer 8 (me). I have installed rust-gnu, mingw, nasm and cmake through scoop. I run Windows 10 Enterprise LTSC, Build 19044. If I run `cargo install librespot`, compilation will stop with this error message:
```
  CMake Error at C:/Users/tannenfreund87/scoop/apps/cmake/3.31.0/share/cmake-3.31/Modules/CMakeTestCCompiler.cmake:67 (message):
    The C compiler

      "C:/Users/tannenfreund87/scoop/apps/mingw/current/bin/gcc.exe"

    is not able to compile a simple test program.
```
I get the same error with version 0.5.0, but not with 0.4.2

However, digging into that issue, I set up a fresh VM with Windows 10, also LTSC 19044, I left all the stock settings, didn't install anything but scoop and also installed `rust-gnu, nasm, mingw, cmake`, compilation runs without problems and produces a working executable.

I read somewhere, that an installation of another version, but the one from msys/mingw of `sh.exe` can be a problem. How do I find out if this is the culprit and what other measures can I do to troubleshoot my case?

**Comment by kingosticks** [comment]
I don't think we can help you here with problems compiling aws-lc-rs on Windows. Maybe try the support channels for aws-lc-rs, or even scoop or mingw given this is an environment issue? 

There's also info at https://aws.github.io/aws-lc-rs/requirements/windows.html
